src/winning_check.c

```diff
@@ -0,0 +1,258 @@
+    int max = 0;
+    for (int i = 0; i < numbers_of_player ; ++i) {
+        if (player_score[i] > max) { max = player_score[i];continue; }
+        if (player_score[i] == max) {
+            count += 1;
+            if (how_many_player_with_this_same_amount_of_points < count) { how_many_player_with_this_same_amount_of_points = count; }
+        } else count = 0;
+    }
+    if (how_many_player_with_this_same_amount_of_points == 0) {
+        for (int i = 0; i < numbers_of_player; ++i) {
+            if (player_score[i] == max) {
+                table_of_winners[0] = players[i];
+                how_many_players_win[0]+=1;
+                return;
+            }
+        }
+    }
+    int max_value = 0;
+    short par = 0;
+    if (max == 5) {
+        for (int x = 0; x < numbers_of_player; x++) {
+            if (player_score[x] == max && highest_color_cart[x] > max_value) { max_value = highest_color_cart[x];}
+        }
+        for (int i = 0; i < numbers_of_player; ++i) {
+            if (highest_color_cart[i] == max_value && player_score[i] == max) {
+                table_of_winners[par] = players[i];
+                par += 1;
+                how_many_players_win[0]+=1;
+            }
+        }
+        return;
+    }
+    if (max == 0) {
+        for (int x = 0; x < numbers_of_player; x++) {
+            if (player_score[x] == max && highest_cart_value[x] > max_value) { max_value = highest_cart_value[x]; }
+        }
+        for (int i = 0; i < numbers_of_player; ++i) {
+            if (highest_cart_value[i] == max_value && player_score[i] == max) {
+                table_of_winners[par] = players[i];
+                par += 1;
+                how_many_players_win[0]+=1;
+            }
+        }
+        return;
+    }
+    if (max == 8 || max == 4) {
+        for (int i = 0; i < numbers_of_player; ++i) {
+            if (player_score[i] == max && start_of_sequence[i] > max_value) { max_value = start_of_sequence[i]; }
+        }
+        for (int i = 0; i < numbers_of_player; ++i) {
+            if (start_of_sequence[i] == max_value && player_score[i] == max) {
+                table_of_winners[par] = players[i];
+                par += 1;
+                how_many_players_win[0]+=1;
+            }
+        }
+        return;
+    }
+    if (max==1|| max==3 || max==7){
+        int place;
+        if(max==3){place=2;}
+        if(max==7){place=3;}
+        if(max==1){place=1;}
+        for (int i = 0; i < numbers_of_player; ++i) {
+            if (player_score[i] == max && power_of_pair_carts[i][place] > max_value) { max_value = power_of_pair_carts[i][place]; }
+        }
+        for (int i = 0; i < numbers_of_player; ++i) {
+            if (power_of_pair_carts[i][place] == max_value && player_score[i] == max) {
+                table_of_winners[par] = players[i];
+                par += 1;
+                how_many_players_win[0]+=1;
+            }
+        }
+        return;
+    }
```

Comment:
This fragment is full of copy-paste. It would be better to enclose find-max-indeces in some sort of a function.